protos/tcp/tcp.go

```diff
@@ -135,7 +162,7 @@ func (tcp *Tcp) Process(tcphdr *layers.TCP, pkt *protos.Packet) {
 			// create
 			stream = &TcpStream{id: tcp.getId(), tuple: &pkt.Tuple, protocol: protocol, tcp: tcp}
 			stream.tcptuple = common.TcpTupleFromIpPort(stream.tuple, stream.id)
-			tcp.streamsMap[pkt.Tuple.Hashable()] = stream
+			tcp.putStream(pkt.Tuple.Hashable(), stream)
```

Comment:
I agree, there is definitely a race here (and in the protocol plugins) that would affect the correctness of this code if/when we parallelize the processing of incoming packets as described in #203. This change does not fix that. In fact running the parallel benchmark test with the race detector (`-race`) points out one of them.

I can open a separate issue to track the parallelization enhancement so we don't lose track of some of the comments regarding parallel processing from this issue.